### PR TITLE
Use latest Ubuntu (LTS) image consistenly across GitHub workflow

### PR DESCRIPTION
**Pull request by jnoordsij**
### What does this PR do?

Use Ubuntu 22.04 consistenly across GitHub workflow

### Motivation

I noticed that all but these two jobs were running on the `ubuntu-22.04` image. For consistency and easier future updates, ~~I thought using the same image for all jobs makes more sense~~ all jobs were altered to use the `ubuntu-latest` (LTS) variant.

### More

- [ ] Added/updated tests
- [ ] Added/updated documentation

### Additional Notes

Requires https://github.com/traefik/traefik/pull/10649 to be merged first; then `v2.11` into `v3.0` merge, after which this can be merged.

### CODE REVIEW

File: .github/workflows/codeql.yml [hunk 1]
```diff
@@ -11,7 +11,7 @@ on:
 jobs:
   analyze:
     name: Analyze
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-22.04
```
Comment (ldez) [inline .github/workflows/codeql.yml:14]:
```suggestion
    runs-on: ubuntu-latest
```

File: .github/workflows/test-conformance.yaml [hunk 2]
```diff
@@ -15,7 +15,7 @@ env:
 jobs:
 
   test-conformance:
-    runs-on: ubuntu-20.04
+    runs-on: ubuntu-22.04
```
Comment (ldez) [inline .github/workflows/test-conformance.yaml:18]:
```suggestion
    runs-on: ubuntu-latest
```

**Review by ldez — CHANGES_REQUESTED** [review]
I think we don't need an explicit version.
Also `ubuntu-latest` simplify the maintenance.

**Comment by jnoordsij** [comment]
Yeah fair, using `latest` everywhere also makes sense and is even less of a burden. Theoretically this might cause unprompted breakges when the release moves, but probably slim chances of that and it's not harmful.

Updated to use this everywhere instead.

**Comment by ldez** [comment]
The OS variant name is used by GitHub checks, we will have a problem with `v2.11`, `v3.0`, and `master` branches if the checks don't have the same name.

Maybe the best approach is to change that inside `v2.11` and after we will backport that on the other branches.

**Comment by jnoordsij** [comment]
Created https://github.com/traefik/traefik/pull/10649 to do this on `v2.11`. 

Given that `v2.11` does not have the `test-conformance` job, this PR is still required for `v3.0` after merging that and then merging `v2.11` into `v3.0` for applying the last change here.

**Comment by ldez** [comment]
Closes in favor #10649.

We will add the missing parts during the backport.

**Pull request closed without merging**